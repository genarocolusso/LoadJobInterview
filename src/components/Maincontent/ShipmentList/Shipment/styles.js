import styled from "styled-components";

//tried to use props instead of 
export const Container = styled.div`
  display: grid;
  grid-template-columns: 55px 1fr;
  border: 1px solid var(--border-color);
  background-color: ${(props) =>
    props.selected ? "var(--shipment-bg-selected)" : "var(--shipment-bg)"};
  margin: 1px 0px;
  color: ${(props) =>
    props.selected ? `var(--list-text-selected)` : `var(--list-text)`};
  padding: 10px 4px;
  transition: 0.3s ease-in all;
  cursor: pointer;
  :hover {
    background-color: var(--shipment-bg-selected);
    color: var(--list-text-selected);
  }

  ${(props) =>
    props.status === "canceled" &&
    `
    background-color: var(--disabled-box);
    border:1px solid var(--disabled-box);
    color: var(--disabled-text);
    :hover { 
    background-color: var(--disabled-box);
    color: var(--disabled-text);
  }
  `}


`;
export const Date = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;

  > .month {
    text-transform: uppercase;
    color: ${(props) =>
    props.status === 'canceled' ? "var(--disabled-text)" : "var(--green)"};
    font-size: 14px;
    font-weight: bold;
  }

  > .day {
    font-size: 24px;
    font-weight: bold;
    text-align: center;
  }
`;

export const ShipmentStops = styled.div``;

export const Stop = styled.div`
  margin-bottom: 4px;
  font-weight: 600;
`;
