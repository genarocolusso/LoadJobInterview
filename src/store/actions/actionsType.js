export const SELECT_SHIPMENT = "SELECT_SHIPMENT";
export const GET_SHIPMENTS = "GET_SHIPMENTS";
export const CHANGE_STATUS = "CHANGE_STATUS";
