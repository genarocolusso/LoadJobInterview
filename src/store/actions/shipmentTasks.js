import {
  SELECT_SHIPMENT,
  GET_SHIPMENTS,
  CHANGE_STATUS,
} from "../actions/actionsType";
import axios from "axios";
import _ from "lodash";
export const selectShipment = (data) => {
  return (dispatch) => {
    return axios
      .get(
        `https://api-loadsmart-shipments.netlify.app/shipments/${data.id}.json`
      )
      .then(({ data }) => {
        console.log(data);
        dispatch({
          type: SELECT_SHIPMENT, //
          payload: data,
        });
      })
      .catch(() => {
        //- show error message

        console.log(2);
      });
  };
};

export const getallShipment = () => {
  return (dispatch) => {
    return axios
      .get("https://api-loadsmart-shipments.netlify.app/shipments/all.json")
      .then(({ data }) => {
        const sorted = _.sortBy(data, ["id"]);
        return dispatch({
          type: GET_SHIPMENTS, //
          payload: sorted,
        });
      })
      .catch(() => {
        //- show error message

        console.log(2);
      });
  };
};

export const changeShipmentStatus = (status) => {
  return (dispatch) => {
    return dispatch({
      type: CHANGE_STATUS, //
      payload: status,
    });
  };
};
