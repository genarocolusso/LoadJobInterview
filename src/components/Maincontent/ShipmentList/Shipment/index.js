import React from "react";
import { Date, ShipmentStops, Stop, Container } from "./styles";

import { selectShipment } from "../../../../store/actions/shipmentTasks";
import { useSelector, useDispatch } from "react-redux";

// function to filter date to the layout
import { getDate } from "../../../util";

//Shipment component
// access shipment data to render it.
const Shipment = ({ shipment }) => {
  const dispatch = useDispatch();
  const selectedid = useSelector((state) => state.shipment.selectedShipment.id);

  const selectHandle = (shipment) => {
    dispatch(selectShipment(shipment));
  };

  return (
    <>
      {shipment && (
        <Container
          selected={selectedid === shipment.id}
          status={shipment.status}
          onClick={() => selectHandle(shipment)}
        >
          <Date status={shipment.status}>
            <div className="month">
              {getDate(`${shipment.stops[0].windowStart}`, "MMM")}
            </div>
            <div className="day">
              {getDate(`${shipment.stops[0].windowStart}`, "DD")}
            </div>
          </Date>

          <ShipmentStops>
            {shipment.stops.map((stop, idx) => (
              <Stop key={idx}>
                {stop.city}, {stop.state}
              </Stop>
            ))}
          </ShipmentStops>
        </Container>
      )}
    </>
  );
};

export default Shipment;
