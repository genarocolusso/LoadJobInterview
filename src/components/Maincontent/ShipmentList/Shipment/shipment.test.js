import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import {
  GET_SHIPMENTS,
  SELECT_SHIPMENT,
} from "../../../../store/actions/actionsType";

import * as actions from "../../../../store/actions/shipmentTasks";

import thunk from "redux-thunk";

import configureStore from "redux-mock-store"; 
import Shipment from ".";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialstate = {
  allShipments: [
    {
      id: 2,
      equipmentType: "DRV",
      status: "current",
      stops: [
        {
          city: "Savannah",
          state: "GA",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
        {
          city: "Aurora",
          state: "IL",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
      ],
    },
    {
      id: 4,
      equipmentType: "DRV",
      status: "canceled",
      stops: [
        {
          city: "Savannah",
          state: "GA",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
        {
          city: "New York",
          state: "NY",
          zipcode: "10009",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
      ],
    },
    {
      id: 5,
      equipmentType: "DRV",
      status: "canceled",
      stops: [
        {
          city: "Savannah",
          state: "GA",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
        {
          city: "Aurora",
          state: "IL",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
      ],
    },
    {
      id: 6,
      equipmentType: "RFR",
      status: "canceled",
      stops: [
        {
          city: "Savannah",
          state: "GA",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
        {
          city: "New York",
          state: "NY",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
      ],
    },
    {
      id: 3,
      equipmentType: "DRV",
      status: "current",
      stops: [
        {
          city: "Bronx",
          state: "NY",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
        {
          city: "Chicago",
          state: "IL",
          country: "USA",
          windowStart: "2021-02-26T14:00:00.000Z",
          windowEnd: "2021-02-26T22:00:00.000Z",
        },
      ],
    },
    {
      id: 1,
      equipmentType: "DRV",
      status: "current",
      stops: [
        {
          city: "New York",
          state: "NY",
          country: "USA",
          windowStart: "2021-03-26T14:00:00.000Z",
          windowEnd: "2021-03-26T22:00:00.000Z",
        },
        {
          city: "Miami",
          state: "FL",
          country: "USA",
          windowStart: "2021-03-29T14:00:00.000Z",
          windowEnd: "2021-03-29T22:00:00.000Z",
        },
      ],
    },
  ],
  selectedShipment: {
    id: 2,
    price: 2240,
    shipperRatingScore: 3.2,
    weight: 45000,
    commodity: "Apples",
    equipmentSize: "53",
    equipmentType: "DRV",
    status: "current",
    map: "https://api-loadsmart-shipments.netlify.app/assets/map.png",
    stops: [
      {
        address: "404 W Bay St",
        city: "Savannah",
        state: "GA",
        country: "USA",
        zipcode: "31401",
        windowStart: "2021-02-26T14:00:00.000Z",
        windowEnd: "2021-02-26T22:00:00.000Z",
        startTimeZone: "America/New_York",
        latitude: 40.722453,
        longitude: -74.002819,
        contact: {
          name: "Sakura Asaki",
          phoneNumber: "+12120001122",
        },
      },
      {
        address: "39W585 Jericho Rd",
        city: "Aurora",
        state: "IL",
        zipcode: "60506",
        country: "USA",
        windowStart: "2021-02-26T14:00:00.000Z",
        windowEnd: "2021-02-26T22:00:00.000Z",
        startTimeZone: "America/New_York",
        latitude: 25.775148,
        longitude: -80.193629,
        contact: {
          name: "Tom Lucas",
          phoneNumber: "+13470001122",
        },
      },
    ],
  },
};
describe("My Connected React-Redux Component", () => {
  let store;
  let component;
  const testShipment = {
    id: 2,
    equipmentType: "DRV",
    status: "current",
    stops: [
      {
        city: "Savannah",
        state: "GA",
        country: "USA",
        windowStart: "2021-02-26T14:00:00.000Z",
        windowEnd: "2021-02-26T22:00:00.000Z",
      },
      {
        city: "Aurora",
        state: "IL",
        country: "USA",
        windowStart: "2021-02-26T14:00:00.000Z",
        windowEnd: "2021-02-26T22:00:00.000Z",
      },
    ],
  };
  beforeEach(() => {
    store = mockStore({
      shipment: initialstate,
    });

    component = renderer.create(
      <Provider store={store}>
        <Shipment shipment={testShipment} />
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  //jest --updateSnapshot  , flag to regenerate snapshots. can be changed on package
});
