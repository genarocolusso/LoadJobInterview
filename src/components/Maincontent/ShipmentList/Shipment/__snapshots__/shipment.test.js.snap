// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="sc-bdfBwQ bMIFzs"
  onClick={[Function]}
  selected={true}
>
  <div
    className="sc-gsTCUz jTLTdm"
  >
    <div
      className="month"
    >
      Feb
    </div>
    <div
      className="day"
    >
      26
    </div>
  </div>
  <div
    className="sc-dlfnbm"
  >
    <div
      className="sc-hKgILt jXsHHf"
    >
      Savannah
      , 
      GA
    </div>
    <div
      className="sc-hKgILt jXsHHf"
    >
      Aurora
      , 
      IL
    </div>
  </div>
</div>
`;
